src/core/prometheus.cc

```diff
@@ -147,9 +147,9 @@ static std::string to_str(const seastar::metrics::impl::metric_value& v) {
     case seastar::metrics::impl::data_type::GAUGE:
         return std::to_string(v.d());
     case seastar::metrics::impl::data_type::COUNTER:
-        return std::to_string(v.i());
-    case seastar::metrics::impl::data_type::DERIVE:
         return std::to_string(v.ui());
+    case seastar::metrics::impl::data_type::DERIVE:
+        return std::to_string(v.i());
```

Comment:
You are right. The difference between COUNTER and DERIVE in collectd was already documented long ago in Scylla's docs/metrics.md, but it appears the implementation never matched the explanation. In any case, in Scylla, all COUNTER and DERIVE are positive, so this different didn't matter.